src/_fixErroneousExpansionMdx.ts

```diff
@@ -0,0 +1,62 @@
+import _get from "lodash/get";
+import _set from "lodash/set";
+import _last from "lodash/last";
+import produce from "immer";
+import {
+  MdxSelect,
+  MdxDrillthrough,
+  isMdxDrillthrough,
+  getLevels,
+} from "@activeviam/mdx";
+import { Cube } from "@activeviam/data-model";
+
+import { _doesCrossjoinRepresentAnExpandedMember } from "./_doesCrossjoinRepresentAnExpandedMember";
+import { _doesCrossjoinYieldAllCombinationsOfMembers } from "./_doesCrossjoinYieldAllCombinationsOfMembers";
+import {
+  ErroneousUnionNode,
+  _findErroneousUnionNode,
+} from "./_findErroneousUnionNode";
+
+/**
+ * Returns an Mdx which yields the same results as the input `mdx`.
+ * But the returned mdx does not have the typical useless (and dangerous) parts of queries created by collapsing then re-expanding a member in ActiveUI 4.
+ * See https://support.activeviam.com/jira/browse/UI-6692
+ */
+export function _fixErroneousExpansionMdx(
+  mdx: MdxSelect | MdxDrillthrough,
+  cube: Cube
+): MdxSelect | MdxDrillthrough {
+  if (isMdxDrillthrough(mdx)) {
+    return mdx;
+  }
+
+  return produce<MdxSelect>(mdx, (draft) => {
+    draft.axes.forEach((axis) => {
+      const levelsOnAxis = getLevels(axis, { cube });
+
+      let erroneousUnionNode:
+        | { path: (string | number)[]; match: ErroneousUnionNode }
+        | undefined = undefined;
```

Comment:
```suggestion
let erroneousUnionNode: ReturnType<typeof _findErroneousUnionNode>
```

Reply:
I saw you started introducing this syntax in ActiveUI and hesitated making a comment. Let's talk about it here.

I am not in favor of seeing this syntax grow. I have 2 arguments in particular:

1. I do not believe that `ReturnType` is well handled by [react-api-documenter](https://github.com/activeviam/react-api-documenter), so accepting to use it either means:
  a. changing react-api-documenter to make it support it
  b. introducing a convention that using it for internal parts of ActiveUI is fine, but using it for exported functions (or types transitively used by exported functinos) is not.
  c. live with the fact that parts of our documentation will become less useful.

In my previous, I went with b, but I am not satisfied with it: if it grows, I know that I will spend time mentioning it (in average multiple times) to each individual author, and introduce avoidable friction in reviews.

2. Is it really more helpful to the reader? Like, is it not better to have the function actually say what it requires, instead of telling you _"go and look to the definition of this other function in order to know what I take in"_. Think about a future where many types are defined this way. It means that you will generally not only have to go look at a separate function's signature, but actually even have to do it several times in a row, if functionA depends on the return type of functionB, which depends on the return type of ...

Reply:
I agree with the last part as a general guideline , however I can think of cases where a given type definition will be repeated several times across the codebase , in that case I would argue that such a redundancy will become difficult to handle.

In this particular case , I think of `erroneousUnionNode` as a simple object that need no type definition of its own, since it's the result of what ever the function returns ( the need for explicitly typing the object comes from the style of the while loop chosen later on ).

Reply:
I think that there is a middleground which satisfies us both:
- avoid ReturnType
- avoid redefining the same type multiple times.

This can be achieved by simply putting a name on every type. I will go for it here.

Reply:
works for me.